Mathlib/Topology/MetricSpace/PseudoMetric.lean

```diff
@@ -1883,6 +1883,27 @@ theorem IsCompact.isSeparable {s : Set α} (hs : IsCompact s) : IsSeparable s :=
   .of_subtype s
 #align is_compact.is_separable IsCompact.isSeparable
 
+lemma TotallyBounded.isSeparable {s : Set α} (h : TotallyBounded s) :
+    TopologicalSpace.IsSeparable s:= by
```

Comment:
IMHO, this should be formulated for `[UniformSpace X] [IsCountablyGenerated (𝓤 X)]`

Reply:
Okay, so if I'm not mistaken, these are equivalent, right (see the below, which typechecks for me)? May I ask why you prefer this formulation? The argument that I know uses metric balls, so I figured it would be naturally to formulate it in terms of [PseudoMetricSpace]. I'm happy to change it though! 

```
import Mathlib

open MeasureTheory TopologicalSpace Uniformity Filter

noncomputable instance {α : Type*} [UniformSpace α] [IsCountablyGenerated (𝓤 α)] : PseudoMetricSpace α := pseudoMetrizableSpacePseudoMetric α
noncomputable instance {α : Type*} [PseudoMetricSpace α] : UniformSpace α := PseudoMetricSpace.toUniformSpace
noncomputable instance {α : Type*} [PseudoMetricSpace α] : IsCountablyGenerated (𝓤 α) := by
  exact EMetric.instIsCountablyGeneratedUniformity
```

Reply:
I’m not too familiar with uniformities, so this’ll take a bit of time. I think I know how to approach this, but I’ll have to see how rapidly this goes

Reply:
I'm stuck, with either of these two proofs, as I'm not sure how to derive the necessary information from the uniformities....


```
import Mathlib

open MeasureTheory TopologicalSpace Uniformity Filter

lemma TotallyBounded.isSeparable5 {α : Type*} [UniformSpace α] [i : IsCountablyGenerated (𝓤 α)] {s : Set α} (h : TotallyBounded s) :
    TopologicalSpace.IsSeparable s:= by
  unfold TopologicalSpace.IsSeparable
  obtain ⟨V, hVa, hVb⟩ := UniformSpace.has_seq_basis α
  choose! f hf hfb using (fun n : ℕ => h (V n))
  have hVc : ∀ n, (V n) ∈ 𝓤 α := by
    intro n
    exact HasAntitoneBasis.mem hVa n
  have := fun n => h.exists_subset (hVc n)
  choose! g hg hg' hg'' using this
  use ⋃ n, g n
  constructor
  · apply Set.countable_iUnion
    exact fun i => Set.Finite.countable (hg' i)
  · sorry

lemma TotallyBounded.isSeparable2 {α : Type*} [UniformSpace α] [i : IsCountablyGenerated (𝓤 α)] {s : Set α} (h : TotallyBounded s) :
    TopologicalSpace.IsSeparable s:= by
  unfold TopologicalSpace.IsSeparable
  obtain ⟨x, hx⟩ := isCountablyGenerated_iff_exists_antitone_basis.mp i
  choose! f hf hfb using (fun n : ℕ => h (x n))
  use ⋃ n, f n
  have hnu : ∀ n, x n ∈ 𝓤 α := by
    intro n
    exact HasAntitoneBasis.mem hx n
  have hf := fun n => hf n (hnu n)
  have hfb := fun n => hfb n (hnu n)
  constructor
  · apply Set.countable_iUnion
    exact fun i => Set.Finite.countable (hf i)
  · intro y hy
    sorry
```

Reply:
Fixed!